### PR TITLE
Add Boost and CRM Cards to AAG

### PR DESCRIPTION
**Pull request by cbauerman**
#### Changes proposed in this Pull Request:

- Add new `DashPluginItem` component for managing separate Plugins as cards in the Jetpack AAG Dashboard
- Use `DashPluginItem` to add new CRM and Boost Card to the AAG Dashboard
- Rename "Performance" section to "Performance and Growth"

##### `DashPluginItem` 

| State | Screenshot |
| -------|--------------------------|
| Loading | Screen Shot 2021-12-15 at 15 44 42 |
| Prompt Install/Activate Plugin | Screen Shot 2021-12-15 at 18 43 34 |
| Installing/Activating Plugin | Screen Shot 2021-12-15 at 15 46 08 |
| Manage Plugin | Screen Shot 2021-12-15 at 15 38 05 |

##### Overal Changes

Screen Shot 2021-12-15 at 18 46 24

##### Video Demonstration

https://user-images.githubusercontent.com/2810519/146299579-1f8916db-e17a-470c-bdde-3c4f492ebcab.mov

#### Jetpack product discussion

p1HpG7-dHF-p2

#### Does this pull request change what data or activity we track or use?

No

#### Testing instructions:

1. Navigate to `/wp-admin/admin.php?page=jetpack#/dashboard` on site with branch
2. Test each of the following with both the CRM and Boost Plugins
 a. When the Plugin is not installed it shows the "install" prompt.
 b. Clicking "Install Plugin" correctly installs the plugin.
 c. When the plugin is installed but not activated the "Activate Plugin" prompt is shown.
 d. Clicking "Activate Plugin" correctly activates the plugin.
 e. When the Plugin is installed and activated the "Manage Plugin" prompt is shown
 f. Clicking the "Manage Plugin" plugin navigates to the respective plugin dashboard

### CODE REVIEW

**Review by monsieur-z — DISMISSED** [review]
PR tests well and code looks good 👍 

It's not ideal that after activating or installing one of the performance products, the other card shows the loading message as well, but I'm guessing this must be due to a lack of granularity when fetching plugins data.

File: projects/plugins/jetpack/_inc/client/components/plugin-dash-item/style.scss [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+	.plugin-dash-item__section-header,
+	.plugin-dash-item__content,
+	.dops-banner {
+		width: 100%;
+		margin-bottom: 0;
+	}
+
+	.dops-banner.dops-card {
+		border-left-color: #3895ba; // $studio-wordpress-blue-30
```
Comment (jeherve) [inline projects/plugins/jetpack/_inc/client/components/plugin-dash-item/style.scss:18]: What do you think about importing those colors, from `_inc/client/scss/variables/_colors.scss`?

**Review by jeherve — CHANGES_REQUESTED** [review]
I gave it another try, left some remarks below.

********

In Chrome, I'm seeing a layout issue like the one we had fixed in #22075 

image

********

Design-wise, is it okay to have different patterns for those cards? For example, in the screenshot below we have a card with a toggle in the card header as the CTA, and an "i" icon for more information, and then the new Boost and CRM cards with a CTA button and a "Learn More" link.

*********

I clicked and the CRM plugin was successfully activated, the card got refreshed (saw the 2 new cards flip to "Loading" for a moment), but the Boost status didn't change in the UI, even though the plugin is activated now:

image

However when I clicked the Boost card the CRM card finally got updated at the same time. Not sure why this happened?

https://user-images.githubusercontent.com/426388/147958450-fd47399b-c23b-479b-93aa-697694f3d7f0.mov

**Comment by cbauerman** [comment]
>In Chrome, I'm seeing a layout issue like the one we had fixed in #22075
 
I believe this is because I made adjustments on this pr to address that issue instead of trying to fix the underlying issue. Should be an easy fix.

> Design-wise, is it okay to have different patterns for those cards? For example, in the screenshot below we have a card with a toggle in the card header as the CTA, and an "i" icon for more information, and then the new Boost and CRM cards with a CTA button and a "Learn More" link.

those aspects of the design are straight from p1HpG7-dHF-p2, though I did extrapolate for the design when the plugins are installed.

 > I clicked and the CRM plugin was successfully activated, the card got refreshed (saw the 2 new cards flip to "Loading" for a moment), but the Boost status didn't change in the UI, even though the plugin is activated now [...] However when I clicked the Boost card the CRM card finally got updated at the same time. Not sure why this happened? 

I will investigate this

**Comment by cbauerman** [comment]
> I clicked and the CRM plugin was successfully activated, the card got refreshed (saw the 2 new cards flip to "Loading" for a moment), but the Boost status didn't change in the UI, even though the plugin is activated now [...] However when I clicked the Boost card the CRM card finally got updated at the same time. Not sure why this happened?

This seems to be because the response from the request to GET `/plugins` has not been updated, even after the POST `/plugins` returns a success. I did not experience this in my own testing, but it is not hard to imagine some sort of race condition occurring. I went ahead and added a 2500ms wait between the success and fetching the plugin data.

**Review by jeherve — APPROVED** [review]
This should be good to merge now!

As a note, for a follow-up PR maybe:
- The new component's readme is empty.
- We've tried to use hooks instead of classes for new components; while I think we can keep using classes for new AAG classes, for consistency with the existing cards, but maybe we could switch for `PluginDashItem`?
https://github.com/Automattic/jetpack/blob/e13487d0dbbe4c743739b6f1a3e37c143a8c2614/projects/plugins/jetpack/_inc/client/README.md#L8-L10

**Comment by cbauerman** [comment]
> As a note, for a follow-up PR maybe:
> 
> * The new component's readme is empty.

Whoops :P

> * We've tried to use hooks instead of classes for new components; while I think we can keep using classes for new AAG classes, for consistency with the existing cards, but maybe we could switch for `PluginDashItem`?

I love hooks, I did not realize that was the direction of the plugin ( I was trying to be idiomatic ). I will follow up with a PR to refine `DashPluginItem`

**Pull request merged**